Gems/Atom/RHI/Vulkan/Code/Source/RHI/CommandList.cpp

```diff
@@ -441,23 +441,58 @@ namespace AZ
             AZStd::vector<VkDescriptorSet> descriptorSets;
             descriptorSets.reserve(dispatchRaysItem.m_shaderResourceGroupCount);
 
+            AZStd::array<const ShaderResourceGroup*, RHI::Limits::Pipeline::ShaderResourceGroupCountMax> srgByAzslBindingSlot = { {} };
             for (uint32_t srgIndex = 0; srgIndex < dispatchRaysItem.m_shaderResourceGroupCount; ++srgIndex)
             {
                 const ShaderResourceGroup* srg = static_cast<const ShaderResourceGroup*>(dispatchRaysItem.m_shaderResourceGroups[srgIndex]);
-                descriptorSets.emplace_back(srg->GetCompiledData().GetNativeDescriptorSet());
+                srgByAzslBindingSlot[srg->GetBindingSlot()] = srg;
             }
 
-            // add the bindless DescriptorSet if required by the shader
             const PipelineState& globalPipelineState = static_cast<const PipelineState&>(*dispatchRaysItem.m_globalPipelineState);
             const PipelineLayout& globalPipelineLayout = static_cast<const PipelineLayout&>(*globalPipelineState.GetPipelineLayout());
-
             for (uint32_t descriptorSetIndex = 0; descriptorSetIndex < globalPipelineLayout.GetDescriptorSetLayoutCount(); ++descriptorSetIndex)
             {
-                const auto& srgBitSet = globalPipelineLayout.GetAZSLBindingSlotsOfIndex(descriptorSetIndex);
-                if (srgBitSet[RHI::ShaderResourceGroupData::BindlessSRGFrequencyId])
+                RHI::ConstPtr<ShaderResourceGroup> shaderResourceGroup;
```

Comment:
This code is copy pasted from CommitDescriptorSets. Why not just call into that similar to other overloaded Submit function calls?

Reply:
There are some differences with the raytracing pipeline state that I need to resolve, and some other work that needs to happen at the pass level.  I plan to refactor this so it can call CommitDescriptorSets like the others.